credentials/src/main/java/me/uport/sdk/credentials/Credentials.kt

```diff
@@ -232,9 +339,13 @@ class Credentials(
      *  @param expiresInSeconds _optional_ number of seconds of validity of the JWT. This parameter
      *              is ignored if the [payload] already contains an `exp` field
      */
-    suspend fun signJWT(payload: Map<String, Any>, expiresInSeconds: Long = DEFAULT_JWT_VALIDITY_SECONDS): String {
+    suspend fun signJWT(
+        payload: Map<String, Any>,
+        expiresInSeconds: Long = DEFAULT_JWT_VALIDITY_SECONDS,
+        algorithm: String? = null
+    ): String {
         val normDID = normalizeKnownDID(this.did)
-        val alg = if (normDID.startsWith("did:uport:")) JwtHeader.ES256K else JwtHeader.ES256K_R
+        val alg = algorithm ?: if (normDID.startsWith("did:uport:")) ES256K else ES256K_R
```

Comment:
should we remove the logic to support legacy uport did?

Reply:
It does not hurt at this moment and removing it would be time consuming.
The goal is to not spend extra time supporting legacy, not to remove legacy support intentionally.